portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,36 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. TODO: modify this file to handle comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    ArrayList<String> messageList = new ArrayList<String>(
+        Arrays.asList("The weather is rainy.", "Bring an Umbrella.", "Watch out for big puddles."));
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Charlie!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Convert the java ArrayList<String> data to a JSON String.
+        String jsonMessage = convertToJsonUsingGson(messageList);
```

Comment:
you can avoid passing the messageList as a parameter to this function, because it's a member variable of the class (so the function 'convertToJsonUsingGson' can just access the messageList directly)

Reply:
makes sense